test/integration.sh

```diff
@@ -1,47 +1,89 @@
+    num_containers_exp="${#containers_expected[@]}"
 
-    access_token=$(../redwood/cli/bin/redwood token create)
-    INFO "generated testing access token: ${access_token}"
+    # Crude initial test to verify that at least the minimum number of
+    # containers are running.
+    [ "${#containers[@]}" -ge $num_containers_exp ] ||\
+        (ERROR "Not running the required number of docker containers."\
+		&& exit 1)
 
-    redwood_endpoint=$(cat ../redwood/.env | grep 'base_url=' | sed 's/[^=]*=//')
-    INFO "got redwood endpoint from dcc-ops/redwood/.env: ${redwood_endpoint}"
+    for container_expected in "${containers_expected[@]}"; do
+        counter=0
+        for container in "${containers[@]}"; do
+            if [[ $container_expected = $container ]]; then
+                continue
+            else
+                counter=$((counter+1))
+            fi
+        done
 
-    INFO "doing upload with $(pwd)/manifest.tsv"
-    sudo docker run --rm -it -e ACCESS_TOKEN=${access_token} -e REDWOOD_ENDPOINT=${redwood_endpoint} \
-         -v $(pwd)/manifest.tsv:/dcc/manifest.tsv -v $(pwd)/samples:/samples -v $(pwd)/outputs:/outputs \
-         quay.io/ucsc_cgl/core-client:1.1.0-alpha spinnaker-upload --skip-submit --force-upload /dcc/manifest.tsv
+        if [[ "$counter" -eq $num_containers_exp ]]; then
+            ERROR "One or more docker containers not running."\
+		&& exit 1
+        fi
+    done
 
-    object_id=$(cat outputs/receipt.tsv | tail -n +3 | head -n 1 | cut -f 20)
-    bundle_id=$(cat outputs/receipt.tsv | tail -n +3 | head -n 1 | cut -f 19)
-    INFO "the uploaded bundle with id ${bundle_id} has metadata.json with object id ${object_id}"
+}
+
+function main {
+    # This tests whether all Docker containers are installed and then
+    # hits the DCC Dashboard Host and checks whether it returns a 200
+    # status code.
+
+    cd "$(dirname "${BASH_SOURCE[0]}")"
+    LOG_LEVEL_ALL
+    check_setup
 
-    INFO "downloading metadata.json with object id ${object_id}"
-    sudo mkdir outputs/test_download
-    sudo docker run --rm -it -e ACCESS_TOKEN=${access_token} -e REDWOOD_ENDPOINT=${redwood_endpoint} \
-         -v $(pwd)/outputs:/outputs quay.io/ucsc_cgl/core-client:1.1.0-alpha download ${object_id} /outputs/test_download
+    # Get name of the DCC Dashboard Host.
+    curr_dir=$(pwd)
+    cd ../boardwalk
+    dcc_host=$(grep -o 'dcc_dashboard_service.*' .env | cut -f2- -d=)
+    # Check from environment variable value whether user is white-listed.    
+    whitelist_set=$(grep -o 'email_white.*' .env | cut -f2- -d=)
+    cd "$curr_dir"
 
-    [[ ! -f outputs/test_download/${bundle_id}/metadata.json ]] && printf "%s: no such file" "outputs/test_download/${bundle_id}/metadata.json" | FATAL && exit 1
+    # Check if home page is returned. If so it will not output anything,
+    # otherwise it aborts and prints cURL error code to stdout.
+    curl -fsS "https://$dcc_host" > /dev/null
```

Comment:
What does this display if there is an error? It would be nice if you could display the url that failed.

Reply:
This currently displays 
`
curl: (6) Could not resolve host: mkrause1.ucsc-cgp-dev.org
`
so the URL shows. I can put in code that displays HTTP status codes if it's not 200, is that want you meant?

Reply:
That's not the url, that's the host. There are presumably other errors that can occur; it would be nice to know that it failed on `https:/mkrause1.ucsc-cgp-dev.org`.

Reply:
It now returns HTTP status codes and URL if the home page exists, but status isn't 200. If the host can't be resolved it still returns the same. Not sure how to catch that cURL error code in a variable.
I also added the URL (not only the host) to lines 86 and 89.